tests/test_sessions.py

```diff
@@ -108,6 +108,7 @@ def test_properties(self):
         assert session.bin is runner.venv.bin_paths[0]
         assert session.python is runner.func.python
         assert session.invoked_from is runner.global_config.invoked_from
+        assert session.cache_dir == os.path.join(runner.envdir, ".shared")
```

Comment:
At this point, I haven't done a "giant work", so I consider this is ok for doing the tests.

Reply:
It seeems like `runner.envdir` is None here? Because the tests failed with a related message.